src/util/evented.js

```diff
@@ -71,9 +71,13 @@ class Evented {
      */
     fire(type, data) {
         if (this.listens(type)) {
-
             data = util.extend({}, data, {type: type, target: this});
 
+            // determine if source is loaded when event is fired
+            if (typeof data.isSourceLoaded === "function") {
+                data.isSourceLoaded = data.isSourceLoaded();
+            };
```

Comment:
An alternative to this approach might be to change the semantics of `Evented#setEventedParent()` so that, if the evented parent `data` parameter is a function, it is evaluated during `fire()` rather than during `setEventedParent()`.

On the surface, that seems preferable to me because it avoids the hardcoding here... but it would require a review of `setEventedParent` to make sure we aren't counting on `data` being evaluated immediately anywhere else.